lib/matplotlib/cbook/__init__.py

```diff
@@ -1543,6 +1575,132 @@ def pts_to_prestep(x, *args):
     return steps
 
 
+def pts_to_betweenstep(x, *args):
+    """
+    Convert continuous line to between-steps.
+
+    Given a set of ``N`` edges and ``N - 1`` values padded with Nan to size N,
+    converts to ``2N - 2`` points, which when connected linearly give
+    a step function connecting step edges at a specified value
+
+    Parameters
+    ----------
+    x : array
+        The x location of the step edges. May be empty.
+
+    y1, ..., yp : array
+        y arrays to be turned into steps; must have length as ``x`` +/- 1.
+    Returns
+    -------
+    out : array
+        The x and y values converted to steps in the same order as the input;
+        can be unpacked as ``x_out, y1_out, ..., yp_out``.  If the input is
+        length ``N``, each of these arrays will be length ``2N - 2``. For
+        ``N=0``, the length will be 0.
+
+    """
+    args = np.array(args)
+    step_length = max(2 * max(len(x), len(args[0])) - 2, 0)
+    steps = np.zeros((1 + len(args), step_length))
+
+    def __between__(steps, sl0, sl1, _x, _args):    
+        # Be agnostic whether xlike or ylike
+        if _x.flatten().shape != x.shape:
+            if _x.flatten().shape[-1] == _x.shape[-1]:
+                _x = _x[0]
+        steps[sl0, ::steps.shape[-1]-1] = _x[::_x.shape[-1]-1]
+        steps[sl0, 2::2] = _x[1:-1]
+        steps[sl0, 1:-1:2] = _x[1:-1]
+        steps[sl1, 0::2] = _args
+        steps[sl1, 1::2] = _args      
+        return steps
+
```

Comment:
@jklymak It could be condensed to something like this which now works in either direction, but I don't think this actually improves readability.